src/resources/js/v2/tickets-modal.js

```diff
@@ -0,0 +1,362 @@
+	 *
+	 * @since TBD
+	 *
+	 * @param {object} $form - The form we're updating.
+	 */
+	obj.preFillModalCartForm = function( $form ) {
+		$form.find( tribe.tickets.block.selectors.item ).hide();
+
+		const $tribeTicket = $form.closest( tribe.tickets.block.selectors.container );
+		const $items = $tribeTicket.find( tribe.tickets.block.selectors.item );
+
+		// Override the data with what's in the tickets block.
+		$.each( $items, function( index, item ) {
+			const $blockItem = $( item );
+			const $item = $form.find( '[data-ticket-id="' + $blockItem.attr( 'data-ticket-id' ) + '"]' );
+
+			if ( $item ) {
+				const quantity = $blockItem.find( '.tribe-tickets-quantity' ).val();
+				if ( 0 < quantity ) {
+					$item.fadeIn();
+				}
+			}
+		} );
+
+		obj.appendARFields( $form );
+
+		tribe.tickets.loader.hide( $form );
+	};
+
+	/**
+	 * Init the form pre-fills ( cart and AR forms ).
+	 *
+	 * @since TBD
+	 */
+	obj.initModalFormPreFills = function() {
+		// @todo: Fix how this is handling stuff (use container based).
+		const $tribeTicket = $form.closest( tribe.tickets.block.selectors.container );
+
+
+		tribe.tickets.loader.show( $document );
+		$.when(
+			obj.getData()
+		).then(
+			function( data ) {
+				obj.preFillModalCartForm( $( obj.selectors.cartForm ) );
+
+				if ( data.meta ) {
+					$.each( data.meta, function( ticket ) {
+						const $matches = $tribeTicket.find( '[data-ticket-id="' + ticket.ticket_id + '"]' );
+
+						if ( $matches.length ) {
+							obj.preFillModalMetaForm( data.meta );
+
+							return;
+						}
+					} );
+				}
+
+				// If we didn't get meta from the API, let's fill with sessionStorage.
+				const local = tribe.tickets.data.getLocal();
+
+				if ( local.meta ) {
+					obj.preFillModalMetaForm( local.meta );
+				}
+
+				// @todo: review this and use the tickets.loader.hide
+				window.setTimeout( obj.loaderHide, 500, tribe.tickets.modal.selectors.loader );
+			}
+		);
+
+		tribe.tickets.loader.hide( $document );
+	};
+
+	/**
+	 * Attempts to hydrate a dynamically-created attendee form "block" from sessionStorage data.
+	 *
+	 * @since TBD
+	 *
+	 * @param {number} length The "skip" index.
+	 */
+	obj.maybeHydrateAttendeeBlockFromLocal = function( length ) {
+		$.when(
+			tribe.tickets.data.getData()
+		).then(
+			function( data ) {
+				if ( ! data.meta ) {
+					return;
+				}
+
+				const cartSkip = data.meta.length;
+				if ( length < cartSkip ) {
+					obj.modal.preFillModalMetaForm( data.meta );
+					return;
+				}
+				const $attendeeForm = $( obj.selectors.metaForm );
+				const $newBlocks = $attendeeForm.find( obj.selectors.metaItem ).slice( length - 1 );
+
+				if ( ! $newBlocks ) {
+					return;
+				}
+
+				$newBlocks.find( obj.selectors.metaField ).each(
+					function() {
+						const $this = $( this );
+						const name = $this.attr( 'name' );
+						const storedVal = data[ name ];
+
+						if ( storedVal ) {
+							$this.val( storedVal );
+						}
+					}
+				);
+			}
```

Comment:
Lets use a separate method here.

Reply:
This is not ready for review, pushed so Scott could know where we were at.